state-chain/pallets/cf-chain-tracking/src/migrations/v2.rs

```diff
@@ -0,0 +1,141 @@
+use crate::*;
+use sp_std::marker::PhantomData;
+
+pub struct Migration<T: Config<I>, I: 'static>(PhantomData<(T, I)>);
+
+#[cfg(feature = "try-runtime")]
+use sp_std::prelude::Vec;
+
+impl<T, I> OnRuntimeUpgrade for Migration<T, I>
+where
+	T: Config<I>,
+	I: 'static,
+	ChainState<T::TargetChain>: v1::FromV1,
+{
+	fn on_runtime_upgrade() -> Weight {
+		// Runtime-check: only migrate the Bitcoin TrackedData
+		if T::TargetChain::NAME == cf_chains::Bitcoin::NAME {
+			// Compile-time: `impl v1::FromV1 for ChainState<Chain>`
+			//     should be defined for every `Chain` we use this migration with.
+			let translated_opt =
+				CurrentChainState::<T, I>::translate(|old| old.map(v1::FromV1::from_v1))
+					// XXX: is it okay to panic here? How to signalise an error otherwise?
+					.expect("failed to decode v1-storage");
+			if let Some(translated) = translated_opt {
+				CurrentChainState::<T, I>::put(translated);
+			}
+		}
+		// For the chains other than Bitcoin `v1::FromV1` should be defined as
+		//    something that explodes rather than silently corrupts the data.
+
+		Weight::zero()
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, DispatchError> {
+		Ok(Default::default())
+	}
+
+	#[cfg(feature = "try-runtime")]
+	fn post_upgrade(_state: Vec<u8>) -> Result<(), DispatchError> {
+		Ok(())
+	}
+}
+
+mod v1 {
+	use crate::*;
+
+	pub trait FromV1 {
+		type OldType: Decode;
+		fn from_v1(old: Self::OldType) -> Self;
+	}
+
+	#[derive(Copy, Clone, RuntimeDebug, PartialEq, Eq, Encode, Decode, MaxEncodedLen, TypeInfo)]
+	pub enum Never {}
+
+	macro_rules! impl_unreachable_from_v1_for_chain {
+		($chain: ty) => {
+			impl FromV1 for crate::ChainState<$chain> {
+				type OldType = Never;
+				fn from_v1(_: Self::OldType) -> Self {
+					unreachable!(
+						"We are not supposed to have an instance of {}",
+						core::any::type_name::<Self::OldType>()
+					)
+				}
+			}
+		};
+	}
+	impl_unreachable_from_v1_for_chain!(cf_chains::Ethereum);
+	impl_unreachable_from_v1_for_chain!(cf_chains::Polkadot);
+
+	pub mod btc {
+		use cf_chains::btc::{BitcoinFeeInfo, BitcoinTrackedData};
+
+		use super::FromV1;
+		use crate::*;
+
+		// The following type-aliases and constants are defined here intentionally (as opposed to
+		// being imported). The types and values should correspond to the types and values that were
+		// in effect right before this migration.
+		pub type BtcBlockNumber = u64;
+		pub type BtcAmount = u64;
+
+		const BYTES_PER_KILOBYTE: BtcAmount = 1024;
```

Comment:
this is a general const, no need to dupe this

Reply:
That would require extending the public interface of `cf-chains` without adding much value.

I will if you insist.